k8s/objects.bzl

```diff
@@ -0,0 +1,73 @@
+# Copyright 2017 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""An implementation of k8s_object for interacting with an object of kind."""
+
+load(
+    "@io_bazel_rules_docker//skylib:path.bzl",
+    _get_runfile_path = "runfile",
+)
+
+def _runfiles(ctx, f):
+  return "PYTHON_RUNFILES=${RUNFILES} ${RUNFILES}/%s" % _get_runfile_path(ctx, f)
+
+def _run_all_impl(ctx):
+  ctx.actions.expand_template(
+      template = ctx.file._template,
+      substitutions = {
+          "%{resolve_statements}": ("\n" + ctx.attr.delimiter).join([
+              _runfiles(ctx, exe.files_to_run.executable)
+              for exe in ctx.attr.objects
+          ]),
+      },
+      output = ctx.outputs.executable,
+  )
+
+  runfiles = [obj.files_to_run.executable for obj in ctx.attr.objects]
```

Comment:
Should we be using `depset`s here instead of lists?

Reply:
`ctx.runfiles(files = ...)` expects a `sequence of Files`, so `depset` really only helps with potential deduping, which I'm not super concerned about (I'd guess Bazel does this later anyways?).